### DISCUSSION TITLE
TOC Governance

### DISCUSSION THREAD

**Discussion by ColinEberhardt** [finos/technical-oversight-committee#72] [General]
From discussion with @mindthegab - we're close to the point of formalising the TSC. The draft charter, which was initially circulated with the group 3 months ago, has undergone legal review. There have been no substantive changes, other than the name, which is now "Technical Oversight Committee". 

I've included the most recent draft of the charter below. We'd like everyone in the pilot TSC to take a look, ahead of finalisation on July 7th.

# Technical Oversight Committee (TOC) Charter

## TOC Voting

The TOC would have 12 seats, each having a two year term
- 6 elected by the FINOS Governing Board
- 6 elected by the existing TOC members

Nominations are encouraged from both the community and the Governing Board. Nominees must either be active contributors or an employee of a member organisation. 

Candidates are expected to be:

- Engaged, and active - committing time to both formal meetings and various other activities. They should actively engage with FINOS Projects.
- Knowledgeable and professional
- ~Able to leverage their organisation~ Active advocates for FINOS
- Neutral - first-and-foremost representing the needs of the members
- Be present - at a minimum they are expected to prioritise and attend the TOC meetings.
- Serve a 2 year term on the TOC

The voting process has two rounds:

1. The FINOS Governing Board vote is conducted and 6 individuals elected
2. Following this, the TOC vote to fill the remaining seats

The Governing Board vote is conducted via the voting contacts, with one vote per member. 

The TOC elects a Chair (serving a 2 year term), who will have a seat on the Governing Board. It is their responsibility to act as a liaison between the TOC and GB, and manage the TOC activities effectively. 

There is a minimum participation threshold for TOC members, with the expectation that they attend at least 50% of the ~monthly~ scheduled TOC meetings, over a rolling quarter. If a member fails to meet this threshold, it triggers a discussion with the Chair, who will use their discretion to determine whether that member should release their seat, by making a recommendation to the GB.

TOC elections happen at most twice per calendar year. At the beginning Q1 and Q3 if there are any vacant TOC seats, a vote is held. Note, the composition of 6 seats filled via Governing Board vote and 6 via leadership vote is maintained through his voting process, with the exact seat numbers being determined based on vacancies.

The initial term for existing "pilot TSC" members and Chair who will retain their seat on the TOC will be 1 year.

## TOC Responsibilities

The TOC supports both the [FINOS team](https://www.finos.org/team) and the [Governing Board](https://www.finos.org/governing-board) to provide technical oversight for the projects in the FINOS portfolio.

The key responsibilities of the TOC are tabulated below:

| **TOC Responsibilities**                                                   | **Outcome**                                                                                                           |**Frequency** |
|----------------------------------------------------------------------------| --------------------------------------------------------------------------------------------------------------------- | ------------ |
| **Landscape Ownership**                                                    |                                                                                                                       |              |
| Solicit and recommend cross-project synergies                              | Review project landscape and recommend                                                                                | Quarterly    |
| Establish technical and security standards, conventions and best practice  |                                                                                                                       | Twice a year |
| Provide impartial and knowledge-based input into technical disputes        | Provide technical recommendations and next steps for dispute resolution and community messaging/engagement            | As needed    |
| Review of project health                                                   | Review project activity and recommend actions as necessary                                                            | Quarterly    |
| Review of project alignment with FINOS strategy                            | Recommend new or existing project alignment with strategic goals, as set by the Board                                 | Quarterly    |
| Evolve the overall landscape                                               | Examine the FINOS project landscape and highlight gaps or potential for additions or removals                         | Twice a year |
| Assist FINOS in maintaining life-cycle criteria                            | Review existing criteria and recommend potential changes or assert continuation as is.                                | Annually     |
| Assist FINOS in managing life-cycle transitions                            | Evaluate and respond to submitted proposals on monthly basis                                                          | As needed      |
| **Landscape growth**                                                       |                                                                                                                       |              |
| Review and approve proposed new open source software and open data projects (SIGs and Open Standards will be approved by Governing Board)                                                                   | Review new contribution proposals and provide timely recommendations and feedback                                     | As needed      |
| Assist FINOS in promote new contributions                                  | Engage with community to communicate value of projects and document potential use cases                               | As needed      |
| Assist FINOS in nurturing newly contributed projects | Identify opportunities to associate existing open source projects with the FSI community; recommend new projects      | As needed      |
| **FINOS Strategy**                                                         |                                                                                                                       |              |
| Work with the Governing Board to identify use-cases / areas of interest    | Engage with Executive Sponsors to identify use cases and promote implementation of strategically-significant projects | Quarterly    |
| Provide strategic input based on existing landscape components             | Identify potential strategic direction, based on existing project collateral                                          | Quarterly    |
| Report to the board                                                        | Engage in annual strategy planning; provide a quarterly report of TSC activities and decisions                        | Quarterly    |

**Comment by ColinEberhardt** [comment]
Note, @mindthegab suggested that the community seats were populated by representatives from Active projects. I think this is a very good idea in principle, however, my feeling is that the [current Active projects](https://landscape.finos.org/card-mode?project=active) are not a terribly good representation of the ambitions of FINOS and its members. For now, I think we need cast a wider net.

**Reply by rikoe** [reply]
I have concerns about whether this will work in practice, for the following requirements mentioned:
- Engaged, and active - committing time to both formal meetings and various other activities
- Able to leverage their organisation
- Neutral - first-and-foremost representing the needs of the members

Active project members usually already spend most of their available free (or employer-afforded) time to contribute to their projects. Being part of the TOC may add to their responsibilities/burdens and impact their focus, especially given the participation requirements. Speaking as a former lead maintainer of a large FINOS project - meeting overload and burn-out is very real.

The same type of people who are active on a project may also not be the same type of people who can leverage their organisation. I would propose that for the TOC we involve more senior people at organisations, e.g. managers or sponsors of those who are active in projects.

Finally, people who are active on a particular project may not be neutral towards their own or other FINOS projects.

I think leaders of projects should definitely get a say in who is on the TOC, and should definitely be able to throw their hat in the ring, but only drawing from among them may lead to groupthink and preclude bringing in a multitude of perspectives.

**Comment by jonfreedman** [comment]
I think we should remove "monthly" from the be present and minimum participation threshold references to TOC meetings whilst we're experimenting with cadence. Otherwise this looks good.

@timkpaine : would you be interested in standing as a candidate?

**Reply by timkpaine** [reply]
sure

**Reply by mindthegab** [reply]
Thanks @jonfreedman - I removed monthly - FYI @ColinEberhardt @jgavronsky (we need to update our slides)

**Comment by mindthegab** [comment]
Thanks folks @finos/technical-steering-committee for getting this moving. A reminder that Friday is the final deadline as we'll need to send the board deck out.

A couple of suggestion and the deadline:

1. **Staggering**: the way I read the election / selection process right now is that the whole TOC gets revamped every two years with 12 new people. But then I see the Q1/Q3 election timeframe. I think it should be fine, but would be good to be extra clear that there would never be an option where the whole TOC gets re-elected (well aside from catastrophic situations).

2. **Leadership vote**: the board needs to approve election rules next week and "leadership" is not a defined concept. So I need you to be more clear here: is it all mantainers? is it one per project? that gets murky really quick and I don't want the FINOS staff to be in the middle. I would warmly recommend you consider if the "leadership vote" should instead be the TOC vote, whereby existing members of the TOC (including or not the newly selected ones  the remaining seats. It's still democratic (people can put their candidacy), meritocratic (active contribution or membership requirements) but it doesn't not create 1. a massive overhead in managing a LOT of elections and 2. it's clear and doesn't open to "who actually can vote". 

3. Given the TSC already served for 1 year, for those members who'll stay (aka above the attendance treshold), I recommend an initial term of 1year (I know @ColinEberhardt already agreed to that for his chair role) and in Q3 2024 we can start new elections. Unless I hear objections, I'll go with this proposal to ensure appropriate rotation over time.

4. As I discussed with @ColinEberhardt already, the TOC will be asked to evaluate open source software / open data projects contributions. SIGs and Open Standards will still go to the board, given the strategic / business impact of those and the need for executive buy-in / potential funding before starting a large scale (high cost) effort. I reflected this in the responsibilities (@jgavronsky @maoo FYI)

In either case, **I could use final comment by Friday COB saying this is the final version reviewed and approved by the "pilot TSC". In absence of that, please assume I'll take silent consensus on whatever is found on this page by Friday 2PM PST.**

**Reply by jonfreedman** [reply]
I think we had discussed electing ~6 new people and rolling ~6 of the current membership over for +1 year giving a natural cadence to annual elections and keeping a mix of old and new members. I'm not sure how prescriptive we can/should be regarding our own candidacy extension, I'm personally happy to be voted on for a 1yr instead of 2y term once this is ratified, that's probably for the board to decide?

**Reply by ColinEberhardt** [reply]
Regarding staggering, I think this will happen quite naturally, as @jonfreedman mentioned, shortly we'll have ~6 existing members, plus ~6 new members. With the existing on 1 year term, and new on 2 year term. Also, when someone stand-down, which will happen from time-to-time, their seat will be filled by a newly elected member who servers a 2 year term.

Regarding leadership vote, we originally discussed leadership encompassing SIG leadership, project maintainers and standards maintainers / editors. However, at the time we drafted this, we did acknowledge that this is not a 'fully formed' concept within FINOS just yet. I'd happy to change this to a TOC vote.

**Reply by mindthegab** [reply]
@jonfreedman 
> I'm personally happy to be voted on for a 1yr instead of 2y term once this is ratified, that's probably for the board to decide? 

Yep, it's for the board. Given the Chair will be voted in for 1y, seems consistent that just for the existing members which will remain after the transition. I'll make the changes in the deck if don't get objections. Thumbs up / down / comments welcome, cut off it tomorrow COB PST.

@ColinEberhardt 
>  I'd happy to change this to a TOC vote.

I think you guys should make the decision. Personally, as much as I like the leadership vote option from a democratic standpoint, in practice  I think TOC vote would be much easier to manage and in my experience less open to confusion/bias/disputes, at least at this stage. The TOC could alway propose to the board a change at a later stage as the community mature. I'll let you make the change if there's consensus. 

If you end up no changing it's fine, but in that case I'd ask that the TSC takes an additional responsibility of ultimate decision of "who's a maintainer" (everyone with github write access? someone who made more than XX PRs?), "what's the SIG leadership" (one Chair per SIG? all the Chairs of a SIG?), etc.

**Reply by ColinEberhardt** [reply]
> > I'd happy to change this to a TOC vote.

> I think you guys should make the decision.

I've updated the text to reflect this change

**Comment by ColinEberhardt** [comment]
@mindthegab I've made some very minor updates (you can see the parts I've struck out), based on the feedback from @jonfreedman  and @rikoe

**Comment by mindthegab** [comment]
@ColinEberhardt when the text is final - I recommend some sends an email to the TSC list in case folks are not following - while not a blocker, because this is the place where the convo is happening, maybe a nice courtesy.

**Comment by mindthegab** [comment]
Thanks @ColinEberhardt - I've only made a minor edit (for consistency) reflect the 6 member are elected by the TOC (not the Community at large) and to crystalize the decision of a 1 year term for existing pilot TSC members and Chair who will retain their seat after the transition.

On to the board deck now, and looking forward to formalize this!

**Reply by brunodom** [reply]
IMHO, some details should be clarified, as I just did similar work in the OpenSSF project (SLSA). Can actual members go to reelection? if so, can the actual member vote on him/herself? Is there a limit to candidates from the community or should we state that candidates should have FINOS member's references so that we can limit the number of candidates? In another group, we were concerned that we may have at some point dozen of candidates and the election process became overwhelming. 

We may not need to work on the details of the voting process at this moment, but at some point, it should revisit the TOC Governance to go over these "details".

**Reply by mindthegab** [reply]
Sure @brunodom - more than happy for you to take the lead and define these and I'll be happy to bring to the GB. It will need another board approval since the Governing Board to change / update composition rules. 

Not my call, but at this point seems way too detailed, since we've been working on this for months and nobody raise these issues in a timely fashion. 

FWIW, my interpretation is that since not specified, all those things you mentioned are allowed.
